playground/static/index.html

```diff
@@ -66,17 +67,64 @@
 
 
         function updateOutput(input) {
+            // Clear the errors
+            errorLog.innerText = "";
+            errorPrompt.style.display = "none";
+
             let pre = document.createElement("pre")
             switch (selector.value) {
-                case "ast":
-                    pre.innerText = ast(input)
-                    break;
-                case "transpile":
-                    pre.innerText = transpile(input);
-                    break;
-                case "render":
-                    output.innerHTML = transpile(input);
-                    break;
+                case "ast": {
+                    let result = "";
```

Comment:
Another improvement would be to move the common parts out, `result`, `try-catch` and so on

Reply:
Fixed!